### PR TITLE
Add Wordle game implementation using React, TypeScript, and TailwindCSS

### PR DESCRIPTION
**Pull request by andreuu-tsai**
- Implement the Wordle game with `Wordle.tsx`, including game logic for word guesses, game board rendering, and keyboard interaction.
- Add `evaluateGuess` function to compare guesses against the correct word.
- Implement `Character` and `Word` components to display individual characters and words on the game board.
- Create `WordleKeyboard` component with interactive buttons for each letter.
- Style the game board and keyboard using TailwindCSS.
- Configure state management for tracking guesses, input, and game status (win/loss).
- Set up `resetGame` functionality to restart the game.

### Additional Changes:
- Added GameResultDisplay component to display a message based on the game result (win/lose).
- Replaced alert notifications with the new GameResultDisplay component.
- Updated game logic to store the game result (win/lose) in state and show it on the game over screen.

### CODE REVIEW

**Comment by andreuu-tsai** [comment]
I've added a new commit to display the game result (win/lose) using a new GameResultDisplay component. The game now shows a message when the game is over instead of using alerts. The PR description has been updated to reflect these changes.

File: homework/andrew/wordle/src/components/Wordle.tsx [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+      // todo: check word is in the dictionary
+      setInputValue("");
+      setWords((prev) => {
+        return [...prev, word];
+      });
+    },
+    [answer]
+  );
+
+  const handleKeyPress = useCallback(
```
Comment (EricTsai83) [inline homework/andrew/wordle/src/components/Wordle.tsx:220]: "useCallback will check on each render to determine if a new callback function needs to be created. If your function definition is not time-consuming, avoid using it."

File: homework/andrew/wordle/src/components/Wordle.tsx [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+        <GameResultDisplay answer={answer} gameResult={gameResult!} />
+      ) : null}
+      <GameBoard words={wordsDisplay} />
+      <div className="grid grid-cols-3 items-center w-full">
+        <div className="col-start-2 col-span-1 place-self-center w-full max-w-lg">
+          <WordleKeyboard words={words} handleKeyPress={handleKeyPress} />
+        </div>
+        <button
+          onClick={(e) => {
+            e.preventDefault();
```
Comment (EricTsai83) [inline homework/andrew/wordle/src/components/Wordle.tsx:294]: write `e.preventDefault()` and `e.currentTarget.blur()` in resetGame function

**Review by EricTsai83 — COMMENTED** [review]
1. Please merge it yourself if there are no changes to be made in this version.
2. When you have time, please try to refactor using useReducer.

**Review by EricTsai83 — APPROVED** [review]

**Pull request merged**